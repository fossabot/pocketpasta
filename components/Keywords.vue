<template>
  <ul class="list-inline">
    <strong>{{ label }}:</strong>
    <li v-for="tag in tags" :key="tag" class="list-inline-item tags">
      {{ tag }}
    </li>
  </ul>
</template>

<script>
export default {
  props: {
    label: { type: String, default: 'Tags' },
    tags: {
      type: Array,
    },
  },
  head() {
    return {
      meta: [
        { hid: 'keywords', name: 'keywords', content: this.tags.join(',') },
      ],
    };
  },
};
</script>

<style>
.tags {
  background: #1abc9c;
  border-radius: 2px;
  color: #f5f5f5;
  font-weight: bold;
  padding: 2px 4px;
  margin-bottom: 0.5rem;
}
</style>
