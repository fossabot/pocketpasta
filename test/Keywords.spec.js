import { shallowMount } from '@vue/test-utils';
import Keywords from '@/components/Keywords.vue';

const factory = () =>
  shallowMount(Keywords, {
    propsData: {
      tags: ['html5', 'css3', 'jquery', 'bootstrap4'],
    },
  });

describe('KeyValue', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
