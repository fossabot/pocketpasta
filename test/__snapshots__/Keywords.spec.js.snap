// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValue renders properly 1`] = `
"<ul class=\\"list-inline\\"><strong>Tags:</strong> <li class=\\"list-inline-item tags\\">
    html5
  </li><li class=\\"list-inline-item tags\\">
    css3
  </li><li class=\\"list-inline-item tags\\">
    jquery
  </li><li class=\\"list-inline-item tags\\">
    bootstrap4
  </li></ul>"
`;
